In [1]:
from google.colab import drive
drive.mount('/gdrive')
%cd /gdrive/My Drive/libs

Mounted at /gdrive
/gdrive/My Drive/libs


In [2]:
%matplotlib inline
import numpy as np
import scipy as sp
import scipy.sparse as sps
import matplotlib.pyplot as plt
import random
from scipy import stats
from scipy.optimize import fmin
import pandas as pd

In [3]:
from Data_manager.split_functions.split_train_validation_random_holdout import split_train_in_two_percentage_global_sample

data = pd.read_csv(filepath_or_buffer="data_train.csv",
                   sep=",")
data.columns = ["UserID", "ItemID", "Interaction"]

mapped_id, original_id = pd.factorize(data["UserID"].unique())
user_original_ID_to_index = pd.Series(mapped_id, index=original_id)

mapped_id, original_id = pd.factorize(data["ItemID"].unique())
item_original_ID_to_index = pd.Series(mapped_id, index=original_id)

data["UserID"] = data["UserID"].map(user_original_ID_to_index)
data["ItemID"] = data["ItemID"].map(item_original_ID_to_index)


URM_all = sps.coo_matrix((data["Interaction"].values,
                          (data["UserID"].values, data["ItemID"].values)))

URM_all.tocsr()

URM_train_validation, URM_test = split_train_in_two_percentage_global_sample(URM_all, train_percentage = 0.8)
URM_train, URM_validation = split_train_in_two_percentage_global_sample(URM_train_validation, train_percentage = 0.8)

URM_all, URM_train_validation, URM_train, URM_validation, URM_test

(<12638x22222 sparse matrix of type '<class 'numpy.float64'>'
 	with 478730 stored elements in COOrdinate format>,
 <12638x22222 sparse matrix of type '<class 'numpy.float64'>'
 	with 382984 stored elements in Compressed Sparse Row format>,
 <12638x22222 sparse matrix of type '<class 'numpy.float64'>'
 	with 306387 stored elements in Compressed Sparse Row format>,
 <12638x22222 sparse matrix of type '<class 'numpy.float64'>'
 	with 76597 stored elements in Compressed Sparse Row format>,
 <12638x22222 sparse matrix of type '<class 'numpy.float64'>'
 	with 95746 stored elements in Compressed Sparse Row format>)

In [18]:
from Recommenders.BaseRecommender import BaseRecommender

class ScoresHybridRecommender(BaseRecommender):
    """ ScoresHybridRecommender
    Hybrid of two prediction scores R = R1*alpha + R2*(1-alpha)

    """

    RECOMMENDER_NAME = "ScoresHybridRecommender"

    '''def __init__(self, URM_train, alpha_R, beta, topK_R, alpha_S, l1_ratio, topK_S):
        super(ScoresHybridRecommender, self).__init__(URM_train)

        self.URM_train = sps.csr_matrix(URM_train)
        self.recommender_1 = RP3betaRecommender(URM_train)
        self.recommender_1.fit(alpha = alpha_R, beta =beta, topK = topK_R,implicit = False, normalize_similarity = True)

        self.recommender_2 = SLIMElasticNetRecommender(URM_train)
        self.recommender_2.fit(alpha = alpha_S, l1_ratio = l1_ratio, topK = topK_S)'''

    def __init__(self, URM_train, recommender1, recommeder2):
          super(ScoresHybridRecommender, self).__init__(URM_train)

          self.URM_train = sps.csr_matrix(URM_train)
          self.recommender_1 = recommender1
          self.recommender_2 = recommeder2


    def fit(self, alpha = 0.5):
        self.alpha = alpha


    def _compute_item_score(self, user_id_array, items_to_compute):

        # In a simple extension this could be a loop over a list of pretrained recommender objects
        item_weights_1 = self.recommender_1._compute_item_score(user_id_array)
        item_weights_2 = self.recommender_2._compute_item_score(user_id_array)

        item_weights = item_weights_1*self.alpha + item_weights_2*(1-self.alpha)

        return item_weights

In [20]:
from Recommenders.GraphBased.RP3betaRecommender import RP3betaRecommender

RP3beta = RP3betaRecommender(URM_train)
RP3beta.fit(alpha = 0.31583032356728813, beta=0.1936831182299878, topK=29, implicit = False, normalize_similarity = True )

RP3betaRecommender: URM Detected 444 ( 3.5%) users with no interactions.
RP3betaRecommender: URM Detected 338 ( 1.5%) items with no interactions.
RP3betaRecommender: Similarity column 22222 (100.0%), 5494.24 column/sec. Elapsed time 4.04 sec


In [9]:
from Recommenders.SLIM.SLIMElasticNetRecommender import SLIMElasticNetRecommender

SLIM_MSE = SLIMElasticNetRecommender(URM_train)
SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237)

SLIMElasticNetRecommender: URM Detected 444 ( 3.5%) users with no interactions.
SLIMElasticNetRecommender: URM Detected 338 ( 1.5%) items with no interactions.
SLIMElasticNetRecommender: Processed 12917 (58.1%) in 5.00 min. Items per second: 43.05
SLIMElasticNetRecommender: Processed 22222 (100.0%) in 8.70 min. Items per second: 42.58


#Try Hyperparameter tuning with Optuna

In [10]:
 pip install optuna

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 413.4/413.4 kB 2.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 233.4/233.4 kB 17.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.6/78.6 kB 11.7 MB/s eta 0:00:00


In [11]:
from Evaluation.Evaluator import EvaluatorHoldout


evaluator_validation = EvaluatorHoldout(URM_validation, cutoff_list=[10])
evaluator_test = EvaluatorHoldout(URM_test, cutoff_list=[10])

EvaluatorHoldout: Ignoring 2608 (20.6%) Users that have less than 1 test interactions
EvaluatorHoldout: Ignoring 2171 (17.2%) Users that have less than 1 test interactions


In [21]:
import optuna
import pandas as pd
from Recommenders.GraphBased.RP3betaRecommender import RP3betaRecommender
from Recommenders.SLIM.SLIMElasticNetRecommender import SLIMElasticNetRecommender

#"alpha": Real(0.1,0.2),    "beta" : Real(0.1,0.2),    "topK" : Integer(10,50)


def objective_function(optuna_trial):


    recommender_instance = ScoresHybridRecommender(URM_train, RP3beta, SLIM_MSE
                                                   #alpha_S = optuna_trial.suggest_float("alpha_S", 0.001, 1.0),
                                                   #alpha_R = optuna_trial.suggest_float("alpha_R", 0.001, 1.0),
                                                   #beta = optuna_trial.suggest_float("beta", 0.001, 1.0),
                                                   #topK_S = optuna_trial.suggest_int("topK_S", 5, 300),
                                                   #topK_R = optuna_trial.suggest_int("topK_R", 5, 100),
                                                   #l1_ratio = optuna_trial.suggest_float("l1_ratio", 1e-6, 1e-4),
                                                   )
    recommender_instance.fit(alpha = optuna_trial.suggest_float("alpha", 0.001, 1.0),
                            )

    result_df, _ = evaluator_validation.evaluateRecommender(recommender_instance)

    return result_df.loc[10]["MAP"]

In [22]:
class SaveResults(object):

    def __init__(self):
        self.results_df = pd.DataFrame(columns = ["result"])

    def __call__(self, optuna_study, optuna_trial):
        hyperparam_dict = optuna_trial.params.copy()
        hyperparam_dict["result"] = optuna_trial.values[0]

        self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)

In [23]:
optuna_study = optuna.create_study(direction="maximize")

save_results = SaveResults()

optuna_study.optimize(objective_function,
                      callbacks=[save_results],
                      n_trials = 100)

[I 2023-12-24 16:56:02,181] A new study created in memory with name: no-name-d5581c4d-b820-4841-8402-8b545ec744fa


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.
EvaluatorHoldout: Processed 10030 (100.0%) in 10.66 sec. Users per second: 941


[I 2023-12-24 16:56:12,878] Trial 0 finished with value: 0.03077180759309407 and parameters: {'alpha': 0.5459566664848782}. Best is trial 0 with value: 0.03077180759309407.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 8.47 sec. Users per second: 1184


[I 2023-12-24 16:56:21,402] Trial 1 finished with value: 0.03034202946715441 and parameters: {'alpha': 0.8266197388726422}. Best is trial 0 with value: 0.03077180759309407.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.32 sec. Users per second: 1371


[I 2023-12-24 16:56:28,740] Trial 2 finished with value: 0.030362171422241426 and parameters: {'alpha': 0.13923338860726825}. Best is trial 0 with value: 0.03077180759309407.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.01 sec. Users per second: 1431


[I 2023-12-24 16:56:35,772] Trial 3 finished with value: 0.030792404532434682 and parameters: {'alpha': 0.40470069631677824}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.21 sec. Users per second: 1391


[I 2023-12-24 16:56:43,005] Trial 4 finished with value: 0.029863469116460425 and parameters: {'alpha': 0.9841094927355926}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.28 sec. Users per second: 1378


[I 2023-12-24 16:56:50,303] Trial 5 finished with value: 0.030088836664609625 and parameters: {'alpha': 0.9038453891288696}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.71 sec. Users per second: 1494


[I 2023-12-24 16:56:57,034] Trial 6 finished with value: 0.030533696529459555 and parameters: {'alpha': 0.7288873072981418}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.28 sec. Users per second: 1377


[I 2023-12-24 16:57:04,338] Trial 7 finished with value: 0.030545407428508444 and parameters: {'alpha': 0.7309632820195817}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.01 sec. Users per second: 1431


[I 2023-12-24 16:57:11,370] Trial 8 finished with value: 0.030502251182959284 and parameters: {'alpha': 0.20702719521560506}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.38 sec. Users per second: 1359


[I 2023-12-24 16:57:18,769] Trial 9 finished with value: 0.029974093592239397 and parameters: {'alpha': 0.9298190995208572}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.32 sec. Users per second: 1370


[I 2023-12-24 16:57:26,122] Trial 10 finished with value: 0.030732117140641797 and parameters: {'alpha': 0.3325960972796348}. Best is trial 3 with value: 0.030792404532434682.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.80 sec. Users per second: 1476


[I 2023-12-24 16:57:32,939] Trial 11 finished with value: 0.030819889537736296 and parameters: {'alpha': 0.5017617878463814}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.41 sec. Users per second: 1353


[I 2023-12-24 16:57:40,374] Trial 12 finished with value: 0.030812332842741888 and parameters: {'alpha': 0.45999149992116983}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.83 sec. Users per second: 1469


[I 2023-12-24 16:57:47,225] Trial 13 finished with value: 0.030779775752109075 and parameters: {'alpha': 0.556621731513647}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.33 sec. Users per second: 1368


[I 2023-12-24 16:57:54,580] Trial 14 finished with value: 0.02995153824241573 and parameters: {'alpha': 0.013576357017547891}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.48 sec. Users per second: 1341


[I 2023-12-24 16:58:02,084] Trial 15 finished with value: 0.030809385336055574 and parameters: {'alpha': 0.44927758944969703}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.86 sec. Users per second: 1462


[I 2023-12-24 16:58:08,965] Trial 16 finished with value: 0.03070063223029355 and parameters: {'alpha': 0.6293003588740509}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.27 sec. Users per second: 1380


[I 2023-12-24 16:58:16,259] Trial 17 finished with value: 0.030653076484831485 and parameters: {'alpha': 0.3055351812942976}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.82 sec. Users per second: 1471


[I 2023-12-24 16:58:23,099] Trial 18 finished with value: 0.030713909857728688 and parameters: {'alpha': 0.6493326676241442}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 16:58:30,488] Trial 19 finished with value: 0.030727753248192158 and parameters: {'alpha': 0.33019900610088027}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.28 sec. Users per second: 1378


[I 2023-12-24 16:58:37,795] Trial 20 finished with value: 0.030791411479846466 and parameters: {'alpha': 0.4851406602435756}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.98 sec. Users per second: 1436


[I 2023-12-24 16:58:44,810] Trial 21 finished with value: 0.030809927360775043 and parameters: {'alpha': 0.42338660669515277}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.40 sec. Users per second: 1356


[I 2023-12-24 16:58:52,230] Trial 22 finished with value: 0.03049391508015659 and parameters: {'alpha': 0.210705934946039}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.86 sec. Users per second: 1463


[I 2023-12-24 16:58:59,108] Trial 23 finished with value: 0.030811901596797125 and parameters: {'alpha': 0.42427012839328154}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.31 sec. Users per second: 1373


[I 2023-12-24 16:59:06,440] Trial 24 finished with value: 0.03077397173558719 and parameters: {'alpha': 0.5841034033421232}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.17 sec. Users per second: 1400


[I 2023-12-24 16:59:13,628] Trial 25 finished with value: 0.030804471506116867 and parameters: {'alpha': 0.5069185311319722}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.00 sec. Users per second: 1434


[I 2023-12-24 16:59:20,651] Trial 26 finished with value: 0.03074647881751596 and parameters: {'alpha': 0.3948762912093562}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1363


[I 2023-12-24 16:59:28,036] Trial 27 finished with value: 0.03064093433983792 and parameters: {'alpha': 0.6870215154768536}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.91 sec. Users per second: 1453


[I 2023-12-24 16:59:34,967] Trial 28 finished with value: 0.03061566491003205 and parameters: {'alpha': 0.2570044553226838}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.34 sec. Users per second: 1366


[I 2023-12-24 16:59:42,335] Trial 29 finished with value: 0.030790980233901796 and parameters: {'alpha': 0.5585408330146509}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.04 sec. Users per second: 1425


[I 2023-12-24 16:59:49,398] Trial 30 finished with value: 0.030816479134026797 and parameters: {'alpha': 0.5003836805358333}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.21 sec. Users per second: 1390


[I 2023-12-24 16:59:56,637] Trial 31 finished with value: 0.03081029134817799 and parameters: {'alpha': 0.4977840948171709}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.45 sec. Users per second: 1346


[I 2023-12-24 17:00:04,111] Trial 32 finished with value: 0.030754585449999473 and parameters: {'alpha': 0.36220715917126145}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.72 sec. Users per second: 1299


[I 2023-12-24 17:00:11,856] Trial 33 finished with value: 0.030790762632736984 and parameters: {'alpha': 0.47319823196283556}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.34 sec. Users per second: 1367


[I 2023-12-24 17:00:19,216] Trial 34 finished with value: 0.030811316051844763 and parameters: {'alpha': 0.602086368949114}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.29 sec. Users per second: 1376


[I 2023-12-24 17:00:26,529] Trial 35 finished with value: 0.030419321400877015 and parameters: {'alpha': 0.7972651685223207}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.99 sec. Users per second: 1436


[I 2023-12-24 17:00:33,545] Trial 36 finished with value: 0.03079615518523826 and parameters: {'alpha': 0.5224708760681376}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.48 sec. Users per second: 1341


[I 2023-12-24 17:00:41,047] Trial 37 finished with value: 0.030719662441247925 and parameters: {'alpha': 0.3885821144870063}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.03 sec. Users per second: 1428


[I 2023-12-24 17:00:48,095] Trial 38 finished with value: 0.030666290810109596 and parameters: {'alpha': 0.2867201309613049}. Best is trial 11 with value: 0.030819889537736296.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.29 sec. Users per second: 1375


[I 2023-12-24 17:00:55,412] Trial 39 finished with value: 0.030826290968365012 and parameters: {'alpha': 0.44333669447735075}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 17:01:02,803] Trial 40 finished with value: 0.03030360110145777 and parameters: {'alpha': 0.12477238418530812}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.06 sec. Users per second: 1420


[I 2023-12-24 17:01:09,894] Trial 41 finished with value: 0.030813104337780547 and parameters: {'alpha': 0.4390092687830237}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.32 sec. Users per second: 1369


[I 2023-12-24 17:01:17,243] Trial 42 finished with value: 0.03081789947617491 and parameters: {'alpha': 0.47030650578311606}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.82 sec. Users per second: 1470


[I 2023-12-24 17:01:24,090] Trial 43 finished with value: 0.030727990631281035 and parameters: {'alpha': 0.6675584442434659}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.42 sec. Users per second: 1352


[I 2023-12-24 17:01:31,535] Trial 44 finished with value: 0.030749734922217806 and parameters: {'alpha': 0.548394241951816}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.23 sec. Users per second: 1387


[I 2023-12-24 17:01:38,794] Trial 45 finished with value: 0.03078299229296261 and parameters: {'alpha': 0.356658774527953}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.01 sec. Users per second: 1432


[I 2023-12-24 17:01:45,829] Trial 46 finished with value: 0.03081270474291435 and parameters: {'alpha': 0.4388819230494785}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.32 sec. Users per second: 1370


[I 2023-12-24 17:01:53,171] Trial 47 finished with value: 0.030787898210131833 and parameters: {'alpha': 0.5291271459490103}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.89 sec. Users per second: 1455


[I 2023-12-24 17:02:00,090] Trial 48 finished with value: 0.030616151545364203 and parameters: {'alpha': 0.7108750248511551}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.35 sec. Users per second: 1364


[I 2023-12-24 17:02:07,465] Trial 49 finished with value: 0.030775562202282343 and parameters: {'alpha': 0.5836636591140357}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.01 sec. Users per second: 1430


[I 2023-12-24 17:02:14,507] Trial 50 finished with value: 0.030505677412208138 and parameters: {'alpha': 0.7574443257056296}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.24 sec. Users per second: 1386


[I 2023-12-24 17:02:21,769] Trial 51 finished with value: 0.03080932203389857 and parameters: {'alpha': 0.46363092549696094}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.37 sec. Users per second: 1362


[I 2023-12-24 17:02:29,161] Trial 52 finished with value: 0.030813907483897757 and parameters: {'alpha': 0.43850372976445334}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.84 sec. Users per second: 1467


[I 2023-12-24 17:02:36,019] Trial 53 finished with value: 0.03077396777920231 and parameters: {'alpha': 0.40119836939372655}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 17:02:43,407] Trial 54 finished with value: 0.0307525795628989 and parameters: {'alpha': 0.6168654267284772}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.98 sec. Users per second: 1437


[I 2023-12-24 17:02:50,415] Trial 55 finished with value: 0.030810334868410925 and parameters: {'alpha': 0.49257900956952416}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.20 sec. Users per second: 1394


[I 2023-12-24 17:02:57,640] Trial 56 finished with value: 0.03075176454762717 and parameters: {'alpha': 0.34158774933736236}. Best is trial 39 with value: 0.030826290968365012.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.29 sec. Users per second: 1376


[I 2023-12-24 17:03:04,957] Trial 57 finished with value: 0.03083431056038257 and parameters: {'alpha': 0.4360593316742256}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.82 sec. Users per second: 1471


[I 2023-12-24 17:03:11,800] Trial 58 finished with value: 0.03066170536011038 and parameters: {'alpha': 0.2884852139547177}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.26 sec. Users per second: 1382


[I 2023-12-24 17:03:19,085] Trial 59 finished with value: 0.03073320514646561 and parameters: {'alpha': 0.39329826461687845}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.80 sec. Users per second: 1475


[I 2023-12-24 17:03:25,909] Trial 60 finished with value: 0.03076709949516562 and parameters: {'alpha': 0.5554303555577836}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.32 sec. Users per second: 1370


[I 2023-12-24 17:03:33,252] Trial 61 finished with value: 0.030823252464827977 and parameters: {'alpha': 0.43380539449762534}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.29 sec. Users per second: 1377


[I 2023-12-24 17:03:40,564] Trial 62 finished with value: 0.030813610755036743 and parameters: {'alpha': 0.46115886882163204}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.80 sec. Users per second: 1474


[I 2023-12-24 17:03:47,392] Trial 63 finished with value: 0.030777714475621103 and parameters: {'alpha': 0.5150410562682093}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.34 sec. Users per second: 1367


[I 2023-12-24 17:03:54,757] Trial 64 finished with value: 0.030758854389213573 and parameters: {'alpha': 0.37544118375134095}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.80 sec. Users per second: 1474


[I 2023-12-24 17:04:01,583] Trial 65 finished with value: 0.030815948978461648 and parameters: {'alpha': 0.4255664486052332}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.37 sec. Users per second: 1361


[I 2023-12-24 17:04:08,977] Trial 66 finished with value: 0.03078584089002856 and parameters: {'alpha': 0.4183853132378724}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.25 sec. Users per second: 1382


[I 2023-12-24 17:04:16,260] Trial 67 finished with value: 0.030737252528130143 and parameters: {'alpha': 0.3204626636471763}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.93 sec. Users per second: 1448


[I 2023-12-24 17:04:23,221] Trial 68 finished with value: 0.0307838864359306 and parameters: {'alpha': 0.4836224014899917}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.46 sec. Users per second: 1345


[I 2023-12-24 17:04:30,703] Trial 69 finished with value: 0.030486686765101745 and parameters: {'alpha': 0.22282059478980026}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.85 sec. Users per second: 1465


[I 2023-12-24 17:04:37,573] Trial 70 finished with value: 0.030779740144645743 and parameters: {'alpha': 0.5863402819960909}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.39 sec. Users per second: 1357


[I 2023-12-24 17:04:44,987] Trial 71 finished with value: 0.030824562028201344 and parameters: {'alpha': 0.43256258593973923}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.15 sec. Users per second: 1403


[I 2023-12-24 17:04:52,160] Trial 72 finished with value: 0.03078740366203012 and parameters: {'alpha': 0.524110774128614}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.00 sec. Users per second: 1433


[I 2023-12-24 17:04:59,189] Trial 73 finished with value: 0.030788677617940064 and parameters: {'alpha': 0.4129629379689431}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 17:05:06,576] Trial 74 finished with value: 0.030822346452705634 and parameters: {'alpha': 0.4693569671530838}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.87 sec. Users per second: 1459


[I 2023-12-24 17:05:13,475] Trial 75 finished with value: 0.030822607574103343 and parameters: {'alpha': 0.4691332480395076}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.34 sec. Users per second: 1367


[I 2023-12-24 17:05:20,838] Trial 76 finished with value: 0.030805298390542926 and parameters: {'alpha': 0.4710740293106627}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.02 sec. Users per second: 1429


[I 2023-12-24 17:05:27,885] Trial 77 finished with value: 0.03076346753390648 and parameters: {'alpha': 0.36605058340920665}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.21 sec. Users per second: 1391


[I 2023-12-24 17:05:35,122] Trial 78 finished with value: 0.03081038234502866 and parameters: {'alpha': 0.44977665369355}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.37 sec. Users per second: 1361


[I 2023-12-24 17:05:42,516] Trial 79 finished with value: 0.02979163303739601 and parameters: {'alpha': 0.9983234768236326}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.74 sec. Users per second: 1487


[I 2023-12-24 17:05:49,283] Trial 80 finished with value: 0.03079429172799093 and parameters: {'alpha': 0.5674114874860378}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 17:05:56,671] Trial 81 finished with value: 0.030812894649385483 and parameters: {'alpha': 0.5012242279838873}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.93 sec. Users per second: 1264


[I 2023-12-24 17:06:04,635] Trial 82 finished with value: 0.03079197328649006 and parameters: {'alpha': 0.5322771768164254}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.01 sec. Users per second: 1431


[I 2023-12-24 17:06:11,675] Trial 83 finished with value: 0.0308136740571938 and parameters: {'alpha': 0.49804124374739467}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 10.96 sec. Users per second: 915


[I 2023-12-24 17:06:22,676] Trial 84 finished with value: 0.03080932203389857 and parameters: {'alpha': 0.4636716923602223}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 10.90 sec. Users per second: 920


[I 2023-12-24 17:06:33,616] Trial 85 finished with value: 0.030751645856082758 and parameters: {'alpha': 0.3460448647574576}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.83 sec. Users per second: 1468


[I 2023-12-24 17:06:40,475] Trial 86 finished with value: 0.030173273829306037 and parameters: {'alpha': 0.8854579886842318}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.30 sec. Users per second: 1373


[I 2023-12-24 17:06:47,802] Trial 87 finished with value: 0.03069844830587633 and parameters: {'alpha': 0.6392918198730925}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.36 sec. Users per second: 1362


[I 2023-12-24 17:06:55,192] Trial 88 finished with value: 0.03078084397600872 and parameters: {'alpha': 0.40684787045245935}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.85 sec. Users per second: 1464


[I 2023-12-24 17:07:02,067] Trial 89 finished with value: 0.030750209688395393 and parameters: {'alpha': 0.381986913211558}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.41 sec. Users per second: 1354


[I 2023-12-24 17:07:09,499] Trial 90 finished with value: 0.030824439380272156 and parameters: {'alpha': 0.44348396368590753}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.83 sec. Users per second: 1469


[I 2023-12-24 17:07:16,353] Trial 91 finished with value: 0.03081418838721955 and parameters: {'alpha': 0.44707489498880193}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.33 sec. Users per second: 1369


[I 2023-12-24 17:07:23,704] Trial 92 finished with value: 0.030781670860434863 and parameters: {'alpha': 0.48348234253703026}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.37 sec. Users per second: 1361


[I 2023-12-24 17:07:31,101] Trial 93 finished with value: 0.030794893098482667 and parameters: {'alpha': 0.5440454944747387}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.89 sec. Users per second: 1455


[I 2023-12-24 17:07:38,026] Trial 94 finished with value: 0.030823252464827977 and parameters: {'alpha': 0.4338144436045154}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.30 sec. Users per second: 1375


[I 2023-12-24 17:07:45,352] Trial 95 finished with value: 0.03082803969045268 and parameters: {'alpha': 0.43243888839531225}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 6.94 sec. Users per second: 1445


[I 2023-12-24 17:07:52,317] Trial 96 finished with value: 0.030819177388469718 and parameters: {'alpha': 0.4261227949549933}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.29 sec. Users per second: 1377


[I 2023-12-24 17:07:59,628] Trial 97 finished with value: 0.030766082704268415 and parameters: {'alpha': 0.3751711501722034}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.20 sec. Users per second: 1393


[I 2023-12-24 17:08:06,861] Trial 98 finished with value: 0.030733185364541536 and parameters: {'alpha': 0.31928822060154616}. Best is trial 57 with value: 0.03083431056038257.


ScoresHybridRecommender: URM Detected 444 ( 3.5%) users with no interactions.
ScoresHybridRecommender: URM Detected 338 ( 1.5%) items with no interactions.


<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


EvaluatorHoldout: Processed 10030 (100.0%) in 7.00 sec. Users per second: 1432


[I 2023-12-24 17:08:13,903] Trial 99 finished with value: 0.03079564876798201 and parameters: {'alpha': 0.40435764180615824}. Best is trial 57 with value: 0.03083431056038257.
<ipython-input-22-650f32a8104f>:10: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self.results_df = self.results_df.append(hyperparam_dict, ignore_index=True)


In [24]:
pruned_trials = [t for t in optuna_study.trials if t.state == optuna.trial.TrialState.PRUNED]
complete_trials = [t for t in optuna_study.trials if t.state == optuna.trial.TrialState.COMPLETE]

print("Study statistics: ")
print("  Number of finished trials: ", len(optuna_study.trials))
print("  Number of pruned trials: ", len(pruned_trials))
print("  Number of complete trials: ", len(complete_trials))

print("Best trial:")
print("  Value Validation: ", optuna_study.best_trial.value)

Study statistics: 
  Number of finished trials:  100
  Number of pruned trials:  0
  Number of complete trials:  100
Best trial:
  Value Validation:  0.03083431056038257


In [25]:
optuna_study.best_trial

FrozenTrial(number=57, state=TrialState.COMPLETE, values=[0.03083431056038257], datetime_start=datetime.datetime(2023, 12, 24, 17, 2, 57, 645601), datetime_complete=datetime.datetime(2023, 12, 24, 17, 3, 4, 956902), params={'alpha': 0.4360593316742256}, user_attrs={}, system_attrs={}, intermediate_values={}, distributions={'alpha': FloatDistribution(high=1.0, log=False, low=0.001, step=None)}, trial_id=57, value=None)

In [26]:
optuna_study.best_trial.params

{'alpha': 0.4360593316742256}

In [27]:
save_results.results_df

result     alpha
0   0.030772  0.545957
1   0.030342  0.826620
2   0.030362  0.139233
3   0.030792  0.404701
4   0.029863  0.984109
..       ...       ...
95  0.030828  0.432439
96  0.030819  0.426123
97  0.030766  0.375171
98  0.030733  0.319288
99  0.030796  0.404358

[100 rows x 2 columns]

In [28]:
recommender_instance = (URM_train + URM_validation)
recommender_instance.fit(**optuna_study.best_trial.params)

result_df, _ = evaluator_test.evaluateRecommender(recommender_instance)
result_df

AttributeError: ignored

#Let's submit

RP3beta.fit(alpha = 0.30045048565736454, beta=0.15900639335692543, topK=34, implicit = False, normalize_similarity = True) + SLIM_MSE.fit(alpha = 0.0033868445794621033, l1_ratio=2.1304972208263965e-05, topK=36) + alpha = 0.5042400731098166. val:3.09473% submit 13.62%

RP3beta.fit(alpha = 0.30045048565736454, beta=0.15900639335692543, topK=34, implicit = False, normalize_similarity = True) + SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237) + alpha:0.42280199553592035 val: 3.09616% submit:13.801%

RP3beta.fit(alpha = 0.3388784009010535, beta=0.177797903368052, topK=30, implicit = False, normalize_similarity = True) + SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237) + alpha = 0.5616855740532924 val: 3.082% submit: 13.766%

RP3beta.fit(alpha = 0.30045048565736454, beta=0.15900639335692543, topK=34, implicit = False, normalize_similarity = True) +
SLIM_MSE.fit(alpha = 0.0030080355721395845, l1_ratio=5.495490764642386e-06, topK=207) +  alpha:0.4451229997552284 + val:3.034% submit:13.8%

RP3beta.fit(alpha = 0.32572640698412575, beta=0.18217576354665624, topK=31, implicit = False, normalize_similarity = True) + SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237) + alpha:0.4420834976325743 val3.146% submit: 13.82%

RP3beta.fit(alpha = 0.31583032356728813, beta=0.1936831182299878, topK=29, implicit = False, normalize_similarity = True ) + SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237) + alpha:0.4360593316742256 val:3.0834% submit:


In [29]:
from Recommenders.GraphBased.RP3betaRecommender import RP3betaRecommender

RP3beta = RP3betaRecommender(URM_all)
RP3beta.fit(alpha = 0.31583032356728813, beta=0.1936831182299878, topK=29, implicit = False, normalize_similarity = True )

RP3betaRecommender: Similarity column 22222 (100.0%), 3919.18 column/sec. Elapsed time 5.67 sec


In [30]:
from Recommenders.SLIM.SLIMElasticNetRecommender import SLIMElasticNetRecommender

SLIM_MSE = SLIMElasticNetRecommender(URM_all)
SLIM_MSE.fit(alpha = 0.0026695828861131796, l1_ratio=9.7058223865738e-05, topK=237)

SLIMElasticNetRecommender: Processed 10610 (47.7%) in 5.00 min. Items per second: 35.36
SLIMElasticNetRecommender: Processed 22222 (100.0%) in 10.00 min. Items per second: 37.04


In [31]:
from Recommenders.GraphBased.RP3betaRecommender import RP3betaRecommender
from Recommenders.SLIM.SLIMElasticNetRecommender import SLIMElasticNetRecommender


In [32]:
#{'alpha_S': 0.0018039635745485064, 'alpha_R': 0.0034513779391265736, 'beta': 0.9111983209661509, 'topK_S': 187, 'topK_R': 93, 'l1_ratio': 9.371084323801284e-05, 'alpha': 0.2838202981991156}.

recommender = ScoresHybridRecommender(URM_all, RP3beta, SLIM_MSE)
recommender.fit(0.4360593316742256)

In [33]:
users_to_recommend_raw = pd.read_csv(filepath_or_buffer="/gdrive/My Drive/libs/data_target_users_test.csv",
                   sep=",",
                  dtype={"user_id": np.int32}).to_numpy()
users_to_recommend = []
for elem in users_to_recommend_raw:
  users_to_recommend.append(elem[0])

len(users_to_recommend)

10882

In [34]:
ratings = pd.read_csv("/gdrive/My Drive/libs/data_train2.csv",
                       sep=",",
                       names=["user_id", "item_id", "ratings"],
                       header=None,
                       dtype={"user_id": np.int64,
                               "item_id": np.int64,
                               "ratings": np.int64})
ratings.shape

(478730, 3)

In [35]:
def preprocess_data(ratings: pd.DataFrame):
    unique_users = ratings.user_id.unique()
    unique_items = ratings.item_id.unique()

    num_users, min_user_id, max_user_id = unique_users.size, unique_users.min(), unique_users.max()
    num_items, min_item_id, max_item_id = unique_items.size, unique_items.min(), unique_items.max()

    print(num_users, min_user_id, max_user_id)
    print(num_items, min_item_id, max_item_id)

    mapping_user_id = pd.DataFrame({"mapped_user_id": np.arange(num_users), "user_id": unique_users})
    mapping_item_id = pd.DataFrame({"mapped_item_id": np.arange(num_items), "item_id": unique_items})

    ratings = pd.merge(left=ratings,
                       right=mapping_user_id,
                       how="inner",
                       on="user_id")

    ratings = pd.merge(left=ratings,
                       right=mapping_item_id,
                       how="inner",
                       on="item_id")

    return ratings


In [36]:
ratings = preprocess_data(ratings)

12638 1 13024
22222 1 22347


In [37]:
users_ids_and_mappings = ratings[ratings.user_id.isin(users_to_recommend)][["user_id", "mapped_user_id"]].drop_duplicates()
items_ids_and_mappings = ratings[["item_id", "mapped_item_id"]].drop_duplicates()
users_ids_and_mappings

user_id  mapped_user_id
0             1               0
1             2               1
2            26              24
3            36              34
4            41              39
...         ...             ...
469974     9966            9677
470088    11525           11196
471697    11994           11649
476929    10065            9773
478118     8646            8401

[10661 rows x 2 columns]

In [38]:
def prepare_submission(ratings: pd.DataFrame, users_to_recommend: np.array, urm_train: sps.csr_matrix):
    users_ids_and_mappings = ratings[ratings.user_id.isin(users_to_recommend)][["user_id", "mapped_user_id"]].drop_duplicates()
    items_ids_and_mappings = ratings[["item_id", "mapped_item_id"]].drop_duplicates()

    mapping_to_item_id = dict(zip(ratings.mapped_item_id, ratings.item_id))


    recommendation_length = 10
    submission = []
    for idx, row in users_ids_and_mappings.iterrows():
        user_id = row.user_id
        mapped_user_id = row.mapped_user_id
        recommendations = recommender.recommend(mapped_user_id,recommendation_length)
        submission.append((user_id, [mapping_to_item_id[item_id] for item_id in recommendations]))

    users_recommended = [elem[0] for elem in submission]

    missing_users = []

    top10 = [2, 4, 1, 7, 3, 6, 8, 9, 15, 20]  #con remove seen = False
    for elem in users_to_recommend:
      if elem not in users_recommended:
        missing_users.append(elem)

    for i in range(len(missing_users)):
      missing_users[i] = (missing_users[i], [mapping_to_item_id[top10[elem]] for elem in range(10)])

    return submission, missing_users

In [39]:
URM_train = URM_all.tocsr()
submission, missing_users = prepare_submission(ratings, users_to_recommend, URM_all)
len(submission), len(missing_users)


(10661, 221)

In [40]:
submission = submission + missing_users
len(submission)

10882

In [41]:
submission.sort()

In [42]:
def write_submission(submissions):
    with open("./Hybrid_12.csv", "w") as f:
        f.write("user_id,item_list\n")
        for user_id, items in submissions:
          f.write(f"{user_id},{' '.join(str(item) for item in items)}\n")



In [43]:
write_submission(submission)